stm32: Move ethernet driver to device tree

erwango opened the pull request:
Ethernet was the last STM32 peripheral driver not using device tree.

- Add ethernet nodes in device tree
- Convert boards to use dts to enabel ethernet
- Update stm32 ethernet driver
- Simplify use of Kconfig symbol ETH_STM32_HAL

erwango committed:
dts/bindings: Add st,stm32-ethernet binding

Add binding for stm32 ethernet controller.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts/arm/st: Add stm32 ethernet nodes

Populate stm32 soc .dtsi files with ethernet nodes

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/ethernet: stm32: Get static boot configuration from dts

Update stm32 ethernet driver to get basic boot configuration from
device tree.
So far information are:
- Label
- Base address
- IRQ number and prio
- Clock settings

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32: Enable ethernet devices through device tree

Update boards that use to declare ethernet support to enable
matching ethernet node.
Similarly to other devices, use dts API to control ethernet pin
settings.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/ethernet: stm32: Enable Kconfig symbol ETH_STM32_HAL using dts

It should not be needed to configure ETH_STM32_HAL from boards.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts: stm32: stm32f769 is a super set of stm32f767

STM32F769 is STM32F767 + DSI

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ABOSTM reviewed (approved):
LGTM

erwango commented:
One BK job received 2h00 timeout while it was running with no failure:

```
...
  | INFO    - 298/351 nucleo_f767zi             tests/net/lib/mqtt_packet/net.mqtt.packet          PASSED (build)
  | INFO    - 299/351 nucleo_f756zg             tests/net/lib/mqtt_packet/net.mqtt.packet          PASSED (build)
  | # Received cancellation signal, interrupting
  | 🚨 Error: The command exited with status -1
```
Will try to launch again with increased timeout duration

erwango commented:
Set as DNM as conflicts with https://github.com/zephyrproject-rtos/zephyr/pull/26226 that should be merged first

jukkar commented on boards/arm/nucleo_f207zg/pinmux.c:
```diff
@@ -22,7 +22,7 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PG9, STM32F2_PINMUX_FUNC_PG9_USART6_RX},
 #endif
-#ifdef CONFIG_ETH_STM32_HAL
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(mac), okay) && CONFIG_NET_L2_ETHERNET
```
It would be better to have `#if DT_NODE_HAS_STATUS(DT_NODELABEL(mac), okay) && defined(CONFIG_NET_L2_ETHERNET)`, could you change these to have the defined() check.

erwango replied:
@jukkar I'd be happy to change, but there's a whole bunch of similar lines across `boards/*/*/pinmux.c` that should need to be aligned if this is really required, and hence, it might be wiser raising a dedicated issue to fix the whole tree.

erwango committed:
boards: nucleo_h747: Fix pinmux conditional configuration

To avoid potential conflicts, pin settings should depend on
subsystem activation.
Fix this.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
